### PR TITLE
libfive: Add all variants, +qt needs qt@5.15.2:+opengl

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Refresh of deps to fix the build and add variants from CMakeLists.txt

### CODE REVIEW

File: var/spack/repos/builtin/packages/libfive/package.py [hunk 1]
```diff
@@ -6,19 +6,25 @@
...
     depends_on('cmake@3.3:', type='build')
     depends_on('boost@1.65:')
     depends_on('eigen@3.3.0:')
     depends_on('libpng')
-    depends_on('qt@5.7:')
-    depends_on('guile@2.2.1:')
+    depends_on('guile@2.2.1:',      when='+guile')
+    # In case the build fails, try enforcing a newer Qt version:
+    depends_on('qt@5.15.2:+opengl', when='+qt')
+    depends_on('python@3:',         type=('build', 'run'))
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/libfive/package.py]: Based on a quick check on the `libfive` github it seems that Python may also be optional: is that true? Or does it not make much sense to omit the Python bindings?

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/libfive/package.py]: @scheibelp Good, I made `python` optional as well, added another variant from the `CMakeLists.txt` file.

As the latter requests newer `cmake`, updated the `depends_on('cmake@...')` as well.

AFAICS, everything should be accounted for now.

**Comment by scheibelp** [comment]
@spackbot re-run pipeline

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Wait a second, this should not have merged: the CI needs to pass. There should have been a block preventing that (i.e. the github UI shouldn't have allowed this to be merged at all).

**Comment by bernhardkaindl** [comment]
@scheibelp I had automerge enabled, it wasn't me merging it, I wasn't even on github  6 minutes ago.

Likely the needed checks had passed already.

**Comment by scheibelp** [comment]
Apologies I wasn't specifically criticizing you: my intent was more to point out that the UI/automerge shouldn't have proceeded with the merge given that the CI check failed. I may have missed that the CI check is not necessary at the moment.

**Pull request merged**